hollow/src/main/java/com/netflix/hollow/api/producer/validation/ValidationStatus.java

```diff
@@ -0,0 +1,151 @@
+ * 
+ * Validation listener uses this to provide visibility into validations that ran.
+ * This aggregates information across multiple validators per run.
+ *
+ */
+public class ValidationStatus {
+    private final long version;
+    private final Status status;
+    private final Throwable throwable;
+    private final HollowProducer.ReadState readState;
+    private final List<ValidatorStatus> validatorStatusList;
+
+	@Generated("SparkTools")
+	private ValidationStatus(Builder builder) {
+		this.version = builder.version;
+		this.status = builder.status;
+		this.throwable = builder.throwable;
+		this.readState = builder.readState;
+		this.validatorStatusList = builder.validatorStatusList;
+	}
+    
+	public ValidationStatus(long version, Status status, Throwable throwable, ReadState readState,
+			List<ValidatorStatus> validatorStatusList) {
+		super();
+		this.version = version;
+		this.status = status;
+		this.throwable = throwable;
+		this.readState = readState;
+		this.validatorStatusList = validatorStatusList;
+	}
+
+	public long getVersion() {
+		return version;
+	}
+
+	public Status getStatus() {
+		return status;
+	}
+
+	public Throwable getThrowable() {
+		return throwable;
+	}
+
+	public HollowProducer.ReadState getReadState() {
+		return readState;
+	}
+
+	public List<ValidatorStatus> getValidatorStatusList() {
+		return validatorStatusList;
+	}
+
+	@Override
+	public String toString() {
+		return "ValidationStatus [version=" + version + ", status=" + status + ", throwable=" + throwable
+				+ ", validatorStatusList=" + validatorStatusList + "]";
+	}
+
+	/**
+	 * Creates builder to build {@link ValidationStatus}.
+	 * @return created builder
+	 */
+	@Generated("SparkTools")
+	public static Builder builder() {
+		return new Builder();
+	}
+
+	/**
+	 * Builder to build {@link ValidationStatus}.
+	 */
+	@Generated("SparkTools")
+	public static final class Builder {
+		private long version;
+		private Status status;
+		private Throwable throwable;
+		private HollowProducer.ReadState readState;
+		private List<ValidatorStatus> validatorStatusList;
+
+		private Builder() {
+			validatorStatusList = new ArrayList<>();
+		}
+
+		public Builder withVersion(long version) {
+			this.version = version;
+			return this;
+		}
+
+		public Builder success() {
+			this.status = Status.SUCCESS;
+			return this;
+		}
+		
+		public Builder fail(Throwable th) {
+			this.status = Status.FAIL;
+			this.throwable = th;
+			return this;
+		}
+
+		public Builder withThrowable(Throwable throwable) {
+			this.throwable = throwable;
+			return this;
+		}
+
+		public Builder withReadState(HollowProducer.ReadState readState) {
+			this.readState = readState;
+			return this;
+		}
+
+		public Builder addValidatorStatus(Throwable th, String message) {
+			Status success = (th == null)? Status.SUCCESS: Status.FAIL;
+			this.validatorStatusList.add(ValidatorStatus.builder().withStatus(success).withMessage(message).withThrowable(th).withVersion(version).build());
```

Comment:
should this method change this.status as well?

Reply:
At this point these two are being set separately. One is set in producer while the other in ValidationStatus. They happen to be same in all cases. Let me see if there is a way to ensure it.

Reply:
Made some changes. More changes to come to make these interactions clearer.